### PR TITLE
Postgres transactions and other fixes

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

- Attempts to solve https://github.com/PostHog/posthog/issues/3218
- Also fixes [this](https://sentry.io/organizations/posthog/issues/2199735318/?project=5592816&query=is%3Aunresolved) and a few other bugs

## Checklist

-   [ ] Updated Settings section in README.md, if settings are affected
-   [ ] Jest tests

### CODE REVIEW

**Comment by mariusandra** [comment]
Finally ready for a look!

Tests should pass yet postgres has been going for ~20sec now. Will restart.

image

**Comment by mariusandra** [comment]
And it's green! 

In the end, this PR contains:
- Code to run transactions in postgres
- Wraps creating persons (+distinct ids) and element groups (+elements) into transactions
- Adds some missing `awaits` from a few tests
- Also fixed a [very edgy bug](https://github.com/PostHog/plugin-server/pull/137/files#diff-42ca1dd74e4c977b47251e70cacf2a225d63bee80a26e0c90108334f510b6588R156) here
- Cleans up some unused vars here and there
- New `onQuery((queryText: string) => void)` helper in tests that's used to count queries. Could be called with `console.log` to print all running queries.

File: src/ingestion/process-event.ts [hunk 1]
```diff
@@ -182,23 +182,28 @@ export class EventsProcessor {
...
                 // Catch race condition where in between getting and creating,
                 // another request already created this person
                 personFound = await this.db.fetchPerson(teamId, distinctId)
             }
         }
-        const updatedProperties: Properties = { ...propertiesOnce, ...personFound!.properties, ...properties }
-        return await this.db.updatePerson(personFound!, { properties: updatedProperties })
+        if (!personFound) {
+            throw new Error(
+                `Could not find person with distinct id "${distinctId}" in team "${teamId}", even after trying to insert them`
```
Comment (Twixes) [inline src/ingestion/process-event.ts:202]: Does seem very edge

**Review by Twixes — APPROVED** [review]

File: src/ingestion/process-event.ts (hunk 1 above)
Comment (mariusandra) [reply src/ingestion/process-event.ts:202]: image

**Pull request merged**